pubsub/google/cloud/pubsub_v1/gapic/publisher_client.py

```diff
@@ -188,6 +216,15 @@ def __init__(
             client_config["interfaces"][self._INTERFACE_NAME]
         )
 
+        # RPC may raise the ServiceUnavailable error, which is a retryable error.
+        # However, we do not want to retry a request if ServiceUnavailable was
+        # raised, because a Pub/Sub service account does not exist - that is an
+        # "invalid grant" error that should fail the request immediately, thus
+        # the patching.
+        for method_name, method_config in copy.copy(self._method_configs).items():
```

Comment:
According to Martelli, just changing the _values_ of a dict while iterating over it is fine ([link](https://stackoverflow.com/a/2315529)).  I nevertheless opted to use `copy.copy()` anyway,  because somebody in the future might copy the pattern and use it in a loop where _not_ using a copy would actually result in an unnecessary bug.

Reply:
It might be clearer to build the value of `self._method_configs` using a dict comprehension, e.g.:

```python
method_configs = client_config["interfaces"][self._INTERFACE_NAME]
self._method_configs = {
    key: _patch_retry_predicate(method_config)
    for method_config in method_configs
}
```

Reply:
I agree, will change it on the other PR.